File: app/src/main/java/com/proxy/shadowsocksr/impl/obfs/AbsObfs.java
package com.proxy.shadowsocksr.impl.obfs;

public abstract class AbsObfs
{
    protected String obfsParam;
    protected final String remoteIP;
    protected final int remotePort;
    protected final int tcpMss;

    public AbsObfs(String usrParamStr, String rmtIP, int rmtPort, int tcpMss)
    {
        obfsParam = usrParamStr;
        remoteIP = rmtIP;
        remotePort = rmtPort;
        this.tcpMss = tcpMss;
    }

    /**
     * If you want to write obfs plugin,<br/>
     * you need write afterEncrypt and beforeDecrypt
     *
     * @param data before process data
     * @return after process data
     */
    public abstract byte[] afterEncrypt(byte[] data);

    /**
     * If you want to write obfs plugin,<br/>
     * you need write beforeEncrypt and afterDecrypt
     *
     * @param data before process data
     * @return after process data
     */
    public abstract byte[] beforeDecrypt(byte[] data, boolean needsendback);
}

File: app/src/main/java/com/proxy/shadowsocksr/impl/plugin/obfs/AbsObfs.kt
package com.proxy.shadowsocksr.impl.plugin.obfs

import java.util.*

abstract class AbsObfs(protected var obfsParam: String, protected val remoteIP: String, protected val remotePort: Int, protected val tcpMss: Int, protected val shareParam: HashMap<String, Any>)
{
    @Throws(Exception::class)
    abstract fun afterEncrypt(data: ByteArray): ByteArray

    @Throws(Exception::class)
    abstract fun beforeDecrypt(data: ByteArray, needsendback: Boolean): ByteArray
}
